Java/Java.sublime-syntax

```diff
@@ -8,19 +8,42 @@ file_extensions:
 scope: source.java
 contexts:
   main:
-    - match: '^\s*(package)\b(?:\s*([^ ;$]+)\s*(;)?)?'
-      scope: meta.package.java
-      captures:
-        1: keyword.other.package.java
-        2: storage.modifier.package.java
-        3: punctuation.terminator.java
-    - match: '^\s*(import)\b(?:\s*([^ ;$]+)\s*(;)?)?'
-      scope: meta.import.java
-      captures:
-        1: keyword.other.import.java
-        2: storage.modifier.import.java
-        3: punctuation.terminator.java
+    - include: package
+    - include: import
     - include: code
+  package:
+    - match: '\bpackage\b'
+      scope: keyword.control.package.java
+      push:
+        - meta_scope: meta.package.java
+        - match: '[\w\.]+'
+          scope: keyword.other.package.java
+        - match: ;
+          scope: punctuation.terminator.java
+          pop: true
+        - match: (?=\n)
+          pop: true
+  import:
+    - match: \bimport\b
+      scope: keyword.control.import.java
+      push:
+        - meta_scope: meta.import.java
+        - match: \bstatic\b
+          scope: storage.modifier.static.java
+          push:
+            - match: '[\w\.]+'
+              scope: keyword.other.static.import.java
```

Comment:
So while it isn't a variable per se, the textmate scopes have traditionally used `variable.` for various identifiers. One of the more popular ones is `variable.function` which is applied to the identifier in a function/method call.

`storage.type` is generally used for built-in data types such as `int`. Often user-defined classes get `support.class` and namespaces `support.namespace` (see PHP for examples). I'm not up on my Java, are imports always classes?

Perhaps we need a new, top-level scope such as `identifier`? Perhaps `identifier.function`, `identifier.generic.path`, `identifier.generic`, `identifier.import`, etc? In that case we'd probably leave `variable.parameter` and `variable.other.readwrite` as-is (or perhaps `variable.other.readwrite` becomes `variable.generic`?).

In many languages we won't really be able to identify all variables reliably, so we may want to recommend color scheme authors not apply highlighting to `identifier.generic`?